assayist/client/query.py

```diff
@@ -44,3 +44,88 @@ def get_container_content_sources(build_id):
         'internal_urls': internal_urls,
         'upstream_urls': upstream_urls
     }
+
+
+def get_current_and_previous_versions(name, type_, version):
+    """
+    Find the current and previous source locations.
+
+    :param str name: the canonical name of the component
+    :param str type_: the canonical type of the component
+    :param str version: the canonical version of the source location
+    :return: a dictionary of all the previous source locations and the current source location
+    :rtype: dict
+    """
+    # TODO: Consider alternative names as well
+    query = """
+        MATCH (:Component {{canonical_name: '{name}', canonical_type: '{type}'}})
+            <-[:SOURCE_FOR]-(:SourceLocation {{canonical_version: '{version}'}})
+            -[:SUPERSEDES*0..]->(sl:SourceLocation)
+        RETURN sl
+    """.format(name=name, type=type_, version=version)
+    results, _ = neomodel.db.cypher_query(query)
+    rv = []
+    for result in results:
+        rv.append(dict(result[0]))
+    return rv
+
+
+def get_container_built_with_sources(source_locations):
+    """
+    Match container builds that used the input source locations to build the content in the image.
+
+    :param list source_locations: a list of source location dictionaries to match against
+    :return: a list of affected container build Koji IDs
+    :rtype: list
+    """
+    if not source_locations or not isinstance(source_locations, list):
```

Comment:
It only really needs to be an iterable I guess? In Python 3, `isinstance(source_locations, collections.Iterable)` follows duck-typing rules and just looks for the methods that implement iteration.